streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -947,127 +665,428 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
     that: ZSink[R1, E1, A00, A1, C]
   )(f: (B, C) => D)(implicit ev: A00 =:= A1, ev2: A1 =:= A00): ZSink[R1, E1, A00, A1, D] =
     zip(that).map(f.tupled)
-
 }
 
 object ZSink extends ZSinkPlatformSpecific {
 
-  implicit class InputRemainderOps[R, E, A, B](val sink: ZSink[R, E, A, A, B]) extends AnyVal {
+  implicit class InputRemainderOps[R, E, A, B](private val sink: ZSink[R, E, A, A, B]) {
```

Comment:
`extends AnyVal`

Reply:
Can't work because of the inner classes that are defined. Not a big loss.

Reply:
Sorry, can you point me to the defined classes? I can't find anything.

Reply:
`case class State` in `collectAllN`

Reply:
Yes, thanks.